File: M3Tools/Controls/Orders/QuantityNudCtrl.vb
Public Class QuantityNudCtrl
	Public Event QuantityChanged(newQuantity As Integer)

	<ComponentModel.DefaultValue(0)>
	Public Property MinimumValue As Integer
		Get
			Return CInt(nud_Quantity.Minimum)
		End Get
		Set(value As Integer)
			nud_Quantity.Minimum = If(value < 0, 0, value)
		End Set
	End Property

	<ComponentModel.DefaultValue(Integer.MaxValue)>
	Public Property MaximumValue As Integer
		Get
			Return CInt(nud_Quantity.Maximum)
		End Get
		Set(value As Integer)
			nud_Quantity.Maximum = If(value > Integer.MaxValue, Integer.MaxValue, value)
		End Set
	End Property

	<ComponentModel.DefaultValue("Quantity")>
	Public Property Label As String
		Get
			Return lbl_Quantity.Text
		End Get
		Set(value As String)
			lbl_Quantity.Text = value
		End Set
	End Property

	<ComponentModel.DefaultValue(0)>
	Property Quantity As Integer
		Get
			Return CInt(nud_Quantity.Value)
		End Get
		Set(value As Integer)
			nud_Quantity.Value = value
		End Set
	End Property

	Private Sub QuantityValueChanged(sender As Object, e As EventArgs) Handles nud_Quantity.ValueChanged
		RaiseEvent QuantityChanged(Quantity)
	End Sub
End Class

File: M3Tools/Controls/Orders/QuantityNudCtrl.cs
using System;

namespace SPPBC.M3Tools
{
	public partial class QuantityNudCtrl
	{
		/// <summary>
		/// The minimum value allowed
		/// </summary>
		[System.ComponentModel.DefaultValue(0)]
		public int MinimumValue
		{
			get => (int)Math.Round(nud_Quantity.Minimum);
			set => nud_Quantity.Minimum = value < 0 ? 0 : value;
		}

		/// <summary>
		/// The maximum value allowed
		/// </summary>
		[System.ComponentModel.DefaultValue(int.MaxValue)]
		public int MaximumValue
		{
			get => (int)Math.Round(nud_Quantity.Maximum);
			set => nud_Quantity.Maximum = value > int.MaxValue ? int.MaxValue : value;
		}

		/// <summary>
		/// The label shown with the control
		/// </summary>
		[System.ComponentModel.DefaultValue("Quantity")]
		public string Label
		{
			get => lbl_Quantity.Text;
			set => lbl_Quantity.Text = value;
		}

		/// <summary>
		/// The quantity value selected
		/// </summary>
		[System.ComponentModel.DefaultValue(0)]
		public int Quantity
		{
			get => (int)Math.Round(nud_Quantity.Value);
			set => nud_Quantity.Value = value;
		}

		/// <summary>
		/// 
		/// </summary>
		public QuantityNudCtrl()
		{
			InitializeComponent();
		}
	}
}
